src/near-bindgen.ts

```diff
@@ -89,6 +89,50 @@ export function call({
   };
 }
 
+/**
+ * The interface that a middleware has to implement in order to be used as a middleware function/class.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+interface Middleware<Arguments extends Array<any>> {
```

Comment:
Can we make Middleware a `type = Function`, as we don't have other member than `call`. And let user call it as:
```js
@middleware((args) => near.log(args))
```

Reply:
@ailisp I can create a commit for this. It's not difficult.

Reply:
@ailisp This is addressed [here](https://github.com/near/near-sdk-js/pull/263/commits/79a9ecbdb3508e2d753dc852ff880c9bcc865c85).